### DISCUSSION TITLE
glos gl error in context

### DISCUSSION THREAD

**Discussion by DorianGrayCodes** [cogentcore/core#429] [Q&A] [answered]
Hello, I am getting three different glos gl errors while running my Goki 3D viewer (based off the example gi3dviewer) and can't figure out what is causing them. While they don't break the viewer, I output important information to the terminal and sometimes I get these errors in such quantities it hides the information I am trying to print.

I get the following errors:

- glos gl error in context: Uniform SetValue: mvp type: {Float32 0 3}:
- glos gl error in context: Uniform SetValue: AmbLights type: {Float32 3 0}:
- glos gl error in context: sld render:

These errors are all followed by:

502 = INVALID_OPERATION: Given when the set of state for a command is not legal for the parameters given to that command. It is also given for commands where combinations of parameters define what the legal parameters are.

I get these often when opening a new window, and sometimes when trying to rotate or interact with my rendered model. They are consistent when opening new windows I've created, for example when opening up a dialog window with checkboxes, I always get the following in the terminal:

2020/12/30 15:44:54 glos gl error in context: Uniform SetValue: AmbLights type: {Float32 3 0}:
	502 = INVALID_OPERATION: Given when the set of state for a command is not legal for the parameters given to that command. It is also given for commands where combinations of parameters define what the legal parameters are.
2020/12/30 15:44:54 glos gl error in context: sld render:
	502 = INVALID_OPERATION: Given when the set of state for a command is not legal for the parameters given to that command. It is also given for commands where combinations of parameters define what the legal parameters are.
2020/12/30 15:44:54 glos gl error in context: sld render:
	502 = INVALID_OPERATION: Given when the set of state for a command is not legal for the parameters given to that command. It is also given for commands where combinations of parameters define what the legal parameters are.
2020/12/30 15:44:54 glos gl error in context: sld render:
	502 = INVALID_OPERATION: Given when the set of state for a command is not legal for the parameters given to that command. It is also given for commands where combinations of parameters define what the legal parameters are.
2020/12/30 15:44:54 glos gl error in context: sld render:
	502 = INVALID_OPERATION: Given when the set of state for a command is not legal for the parameters given to that command. It is also given for commands where combinations of parameters define what the legal parameters are.
2020/12/30 15:44:54 glos gl error in context: sld render:
	502 = INVALID_OPERATION: Given when the set of state for a command is not legal for the parameters given to that command. It is also given for commands where combinations of parameters define what the legal parameters are.
2020/12/30 15:44:54 glos gl error in context: sld render:
	502 = INVALID_OPERATION: Given when the set of state for a command is not legal for the parameters given to that command. It is also given for commands where combinations of parameters define what the legal parameters are.
2020/12/30 15:44:54 glos gl error in context: sld render:
	502 = INVALID_OPERATION: Given when the set of state for a command is not legal for the parameters given to that command. It is also given for commands where combinations of parameters define what the legal parameters are.
2020/12/30 15:44:54 glos gl error in context: Uniform SetValue: mvp type: {Float32 0 3}:
	502 = INVALID_OPERATION: Given when the set of state for a command is not legal for the parameters given to that command. It is also given for commands where combinations of parameters define what the legal parameters are.

Sometimes when opening the file select window which came with the gi3dviewer example, I also get one of these errors.

Any advice is appreciated.

Thanks,
Daniel

**Comment by rcoreilly** [comment, marked as answer]
This should be filed as an issue -- what platform is this?  I'm assuming windows -- I just built a release for it and noticed similar issues -- was hoping maybe it was just because my laptop is very old.. will take a look.

**Reply by rcoreilly** [reply]
I filed as #430 and it is now fixed in new version
